examples/whatsapp/whatsapp_bot_server.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+from flask import Flask, request
+from twilio.twiml.messaging_response import MessagingResponse
+
+from mindmeld.components import NaturalLanguageProcessor
+from mindmeld.components.dialogue import Conversation
+from mindmeld import configure_logs
+
+
+class WhatsappBotServer:
+    """
+    A sample server class for Whatsapp integration with any MindMeld application
+    """
+
+    def __init__(self, name, app_path, nlp=None):
+        """
+        Args:
+            name (str): The name of the server.
+            app_path (str): The path of the MindMeld application.
+            nlp (NaturalLanguageProcessor): MindMeld NLP component, will try to load from app path
+              if None.
+        """
+        self.app = Flask(name)
+        if not nlp:
+            self.nlp = NaturalLanguageProcessor(app_path)
+            self.nlp.load()
+        else:
+            self.nlp = nlp
+        self.conv = Conversation(nlp=self.nlp, app_path=app_path)
+        self.logger = logging.getLogger(__name__)
+
+        @self.app.route("/", methods=["POST"])
+        def handle_message():  # pylint: disable=unused-variable
+            incoming_msg = request.values.get('Body', '').lower()
+            resp = MessagingResponse()
```

Comment:
Does this object have to be created at every request, so can it be a state var?

Reply:
Yeah I think you need to create a message response for every request:

```
In [2]: ms = MessagingResponse()

In [3]: a = ms.message()

In [4]: a.body("hello world")
Out[4]: <twilio.twiml.messaging_response.Body at 0x106ebf240>

In [8]: str(a)
Out[8]: '<?xml version="1.0" encoding="UTF-8"?><Message><Body>hello world</Body></Message>'

In [9]: b = ms.message()

In [11]: b.body("hello there")
Out[11]: <twilio.twiml.messaging_response.Body at 0x10592e278>

In [12]: str(b)
Out[12]: '<?xml version="1.0" encoding="UTF-8"?><Message><Body>hello there</Body></Message>'

In [13]: str(ms)
Out[13]: '<?xml version="1.0" encoding="UTF-8"?><Response><Message><Body>hello world</Body></Message><Message><Body>hello there</Body></Message></Response>'
```